log4tango/src/PThreads.cpp

```diff
@@ -53,7 +54,7 @@ std::string get_thread_id (void)
 long thread_id (void) 
 {
 #ifndef __darwin__
-	#ifndef __freebsd__
+	#if (!defined (__FreeBSD__) && !defined( __OpenBSD__) )
 		return static_cast<long>(::pthread_self());
 	#else
 		return reinterpret_cast<long>(::pthread_self());
```

Comment:
This function is used only for logging messages. We may replace any calls to it with a simple call to `std::this_thread::get_id()`. It returns `std::thread::id` which (at least on GNU C++ lib) is just a wrapper around thread's native handle (which happens to be pthread_t as returned from `pthread_self`. When printed to log stream, `std::thread::id` shall yield the same result as printing the numerical thread ID directly.

Reply:
Thanks @mliszcz
This suggestion has been implemented in 59931b9fbc92e375ae2e6c99f1adf779b97df975

Reply:
Thanks @bourtemb! I've just opened an issue to remove this file completely and replace it with standard library threading utilities. Should be simple as it impacts only log4tango codebase.
https://github.com/tango-controls/cppTango/issues/712